src/utils.ts

```diff
@@ -0,0 +1,57 @@
+import * as editorconfig from 'editorconfig';
+import {EditorSettings} from './interfaces/editorSettings';
+import {window, workspace, TextDocument, TextEditor} from 'vscode';
+
+export class Utils {
+
+	/**
+	 * Convert .editorconfig values to vscode editor options
+	 */
+	public static fromEditorConfig(config: editorconfig.knownProps, defaults: EditorSettings): any {
+		return {
+			insertSpaces: config.indent_style ? config.indent_style !== 'tab' : defaults.insertSpaces,
+			tabSize: config.tab_width || config.indent_size || defaults.tabSize
+		};
+	}
+
+	/**
+	 * Convert vscode editor options to .editorconfig values
+	 */
+	public static toEditorConfig(options: EditorSettings) {
+		const result: editorconfig.knownProps = {};
+
+		switch (options.insertSpaces) {
+			case true:
+				result.indent_style = 'space';
+				result.indent_size = Utils.resolveTabSize(options.tabSize);
+				break;
+			case false:
+			case 'auto':
+				result.indent_style = 'tab';
+				result.tab_width = Utils.resolveTabSize(options.tabSize);
+				break;
+		}
+
+		return result;
+	}
+
+	/**
+	 * Convert vscode tabSize option into numeric value
+	 */
+	public static resolveTabSize(tabSize: number|string) {
+		return (tabSize === 'auto') ? 4 : parseInt(tabSize + '', 10);
```

Comment:
I think I see what your argument is now. Because the spec says it's a "whole number," you're saying that we should only be concerned with `.editorconfig` files that follow the spec to the T. I guess I was thinking we could be a bit flexible if it's not difficult to do so. What do you think?